http://svn.php.net/viewvc?view=revision&revision=297236
and another patch to fix "error: ‘zend_class_entry’ has no member named ‘default_properties’"
--- gdchart.c.orig	2006-01-03 15:05:29.000000000 -0600
+++ gdchart.c	2012-06-22 18:35:27.000000000 -0500
@@ -192,7 +192,7 @@
  *
  * Every user visible function must have an entry in gdchart_functions[].
  */
-function_entry gdchart_functions[] = {
+zend_function_entry gdchart_functions[] = {
 	{NULL, NULL, NULL}	/* Must be the last line in gdchart_functions[] */
 };
 /* }}} */
@@ -300,7 +300,11 @@
 
 	ALLOC_HASHTABLE(intern->zo.properties);
         zend_hash_init(intern->zo.properties, 0, NULL, ZVAL_PTR_DTOR, 0);
+#if PHP_VERSION_ID < 50399
 	zend_hash_copy(intern->zo.properties, &class_type->default_properties, (copy_ctor_func_t) zval_add_ref, (void *) &tmp, sizeof(zval *));
+#else
+	object_properties_init(&(intern->zo), class_type);
+#endif
 
 #if ZEND_MODULE_API_NO >= 20050922
 	intern->zo.guards = NULL;
